src/main/java/s3NativeClient/com/amazonaws/s3/S3NativeClient.java

```diff
@@ -199,11 +199,11 @@ public int onResponseBody(ByteBuffer bodyBytesIn, long objectRangeStart, long ob
             }
 
             @Override
-            public void onFinished(int errorCode, int responseStatus, byte[] errorPayload) {
+            public void onFinished(S3FinishedResponseContext context) {
                 CrtS3RuntimeException ex = null;
                 try {
-                    if (errorCode != CRT.AWS_CRT_SUCCESS) {
-                        ex = new CrtS3RuntimeException(errorCode, responseStatus, errorPayload);
+                    if (context.getErrorCode() != CRT.AWS_CRT_SUCCESS) {
```

Comment:
Just throw out ideas: I think we can throw out `CrtRuntimeException` if it's not because the server side reject the request. And only throw out `CrtS3RuntimeException` when the error body is informative (server side rejects our request).